### ISSUE TITLE
Unable to connect to Nexys Video with OpenOCD, Error: IR capture error at bit 2

### ISSUE THREAD

**Issue by alistair23** [lowRISC/opentitan#905]
I am following the steps here: https://github.com/lowRISC/opentitan/blob/master/doc/ug/getting_started_fpga.md#connect-with-openocd-and-debug

I have a USB cable connecting my Linux machine to the `PROG` port on the Nexys Video. I can see the boot ROM print to the screen and I can load applications using `./sw/host/spiflash/spiflash`. When I load an application I see output on both the board UART as well as my local Linux machine.

I run OpenOCD using this command line:

```bash
$ openocd -s util/openocd -f ./util/openocd/interface/nexysvideo-ft2232.cfg
```

When running OpenOCD I see this error:

```
Open On-Chip Debugger 0.10.0+dev-00824-gf93ede540-dirty (2019-11-07-00:35)
Licensed under GNU GPL v2
For bug reports, read
	http://openocd.org/doc/doxygen/bugs.html
none separate

Info : Listening on port 6666 for tcl connections
Info : Listening on port 4444 for telnet connections
Info : clock speed 30000 kHz
Warn : There are no enabled taps.  AUTO PROBING MIGHT NOT WORK!!
Info : JTAG tap: auto0.tap tap/device found: 0x04f5484d (mfg: 0x426 (Google Inc), part: 0x4f54, ver: 0x0)
Warn : AUTO auto0.tap - use "jtag newtap auto0 tap -irlen 2 -expected-id 0x04f5484d"
Error: IR capture error at bit 2, saw 0x3FFFFFFFFFFFFFE5 not 0x...3
Warn : Bypassing JTAG setup events due to errors
Warn : gdb services need one or more targets defined

```

I have tried OpenOCD from the [RISC-V repo](https://github.com/riscv/riscv-openocd), commit: f93ede5401c711e55d9852986aa399c0318efb22.
I have also tried the latest from mainline, commit: ded67990255cc1e63c77832ffd6e6bef9120873d

GDB 8.3.1 then fails with this error:

```
warning: unrecognized item "timeout" in "qSupported" response
Ignoring packet error, continuing...
Remote replied unexpectedly to 'vMustReplyEmpty': timeout
```

**Comment by tjaychen** [comment]
hmm i quickly tried mine, and i did not run into this problem.
My fpga is not head of tree, I'll retry and find out.

Also, are you using tip of tree fpga? Or are you using one of the released
builds? The released builds have almost certainly been tested for OpenOCD
compliance and i would expect things to be okay.

**Comment by towoe** [comment]
Please try the following command

```
$ openocd -s util/openocd -f board/lowrisc-earlgrey-nexysvideo.cfg
```

This will use the board file which sets the interface and the target.

**Comment by tjaychen** [comment]
ah thanks @towoe , i did not notice that that usage was different from the instructions.

**Comment by alistair23** [comment]
Thanks for the help.

I followed the instructions board file but couldn't get it to work, so I changed to what is in the issue.

I swapped back and figure out that it doesn't work with mainline OpenOCD only the RISC-V fork which is why I couldn't get the instructions to work. Using the RISC-V fork fixes the error (although I still see the GDB error).

That's probably worth documenting in the instructions, I'll send a patch to add that.
